src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java

```diff
@@ -40,4 +48,15 @@ public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount co
                 return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
         }
     }
+
+    @GetMapping(value = "/transactions/{transactionId}/company-accounts/{companyAccountId}",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity getCompanyAccount(HttpServletRequest request) {
+
+        CompanyAccountEntity companyAccountEntity = (CompanyAccountEntity) request.getSession()
+            .getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
```

Comment:
This should check that the company account was found successfully and return a 404 if it isn't in the session.